File: CS/GridControlCRUDMVVMInfiniteAsyncSource/IssuesContext/IssuesContext.cs
using System.Data.Entity;

namespace GridControlCRUDMVVMInfiniteAsyncSource {
    public class IssuesContext : DbContext {
        static IssuesContext() {
            Database.SetInitializer(new IssuesContextInitializer());
        }
        public IssuesContext() { }

        protected override void OnModelCreating(DbModelBuilder modelBuilder) {
            base.OnModelCreating(modelBuilder);

            modelBuilder.Entity<Issue>()
                .HasIndex(x => x.Created);

            modelBuilder.Entity<Issue>()
                .HasIndex(x => x.Votes);
        }

        public DbSet<Issue> Issues { get; set; }
        public DbSet<User> Users { get; set; }
    }
}

File: VB/CodeBehind/EntityFramework/LocalData/Issues/IssuesContext.vb
Imports System.Data.Entity

Namespace Issues
    Public Class IssuesContext
        Inherits DbContext

        Shared Sub New()
            Database.SetInitializer(New IssuesContextInitializer())
        End Sub

        Public Sub New()
        End Sub

        Protected Overrides Sub OnModelCreating(ByVal modelBuilder As DbModelBuilder)
            MyBase.OnModelCreating(modelBuilder)
            modelBuilder.Entity(Of Issue)().HasIndex(Function(x) x.Created)
            modelBuilder.Entity(Of Issue)().HasIndex(Function(x) x.Votes)
        End Sub

        Public Property Issues As DbSet(Of Issue)
        Public Property Users As DbSet(Of User)
    End Class
End Namespace

